Userland/DevTools/UserspaceEmulator/SoftFPU.cpp

```diff
@@ -810,8 +866,66 @@ void SoftFPU::FUCOMIP(const X86::Instruction& insn)
     fpu_pop();
 }
 
-void SoftFPU::FTST(const X86::Instruction&) { TODO_INSN(); }
-void SoftFPU::FXAM(const X86::Instruction&) { TODO_INSN(); }
+void SoftFPU::FTST(const X86::Instruction&)
+{
+    // FIXME: maybe check for denormal
+    set_c1(0);
+    if (fpu_isnan(0))
+        // raise #IA?
+        fpu_set_unordered();
+    else {
+        set_c0(fpu_get(0) < 0.);
+        set_c2(0);
+        set_c3(fpu_get(0) == 0.);
+    }
+}
+void SoftFPU::FXAM(const X86::Instruction&)
+{
+    // noexcept!
```

Comment:
What does this mean?

Reply:
this is a notice, that that function cannot raise a FPU Exception, shall I reword it or delete it?